applications/OptimizationApplication/python_scripts/algorithms/algorithm_steepest_descent.py

```diff
@@ -1,129 +1,147 @@
-                root_control_model_part = self.model_parts_controller.GetRootModelPart(control_model_part)
-                OriginalTime = root_control_model_part.ProcessInfo[KM.TIME]
-                root_control_model_part.ProcessInfo[KM.TIME] = self.optimization_iteration
-
-                vtkIO.ExecuteInitializeSolutionStep()
-                if(vtkIO.IsOutputStep()):
-                    vtkIO.PrintOutput()
-                vtkIO.ExecuteFinalizeSolutionStep()
-
-                root_control_model_part.ProcessInfo[KM.TIME] = OriginalTime
-                            
-            Logger.Print("")
-            Logger.PrintInfo("AlgorithmSteepestDescent", "Time needed for current optimization step = ", timer.GetLapTime(), "s")
-            Logger.PrintInfo("AlgorithmSteepestDescent", "Time needed for total optimization so far = ", timer.GetTotalTime(), "s")
-
-        self.FinalizeOptimizationLoop()
-
-    # --------------------------------------------------------------------------
-    def FinalizeOptimizationLoop(self):
-        super().FinalizeOptimizationLoop()
-
-# ==============================================================================
+    
+    def __init__(self, model:Kratos.Model, parameters: Kratos.Parameters, optimization_problem: OptimizationProblem):
+        self.model = model
+        self.parameters = parameters
+        self._optimization_problem = optimization_problem
+
+        self.master_control = MasterControl() # Need to fill it with controls
+        self.__control_param_list = parameters["controls"]
+
+        for control_param in self.__control_param_list.values():
+            control_name = control_param.GetString()
+            control = optimization_problem.GetControl(control_name)
+            self.master_control.AddControl(control)
+
+        settings = parameters["settings"]
+        settings.ValidateAndAssignDefaults(self.GetDefaultParameters()["settings"])
+
+        self.gradient_scaling = settings["gradient_scaling"].GetString()
+        self.echo_level = settings["echo_level"].GetInt()
+
+        ComponentDataView("algorithm", self._optimization_problem).SetDataBuffer(self.GetMinimumBufferSize())
+
+        self.__convergence_criteria = CreateConvergenceCriteria(settings["conv_settings"], self._optimization_problem)
+        self.__line_search_method = CreateLineSearch(settings["line_search"], self._optimization_problem)
+
+        self.__objective = StandardizedObjective(parameters["objective"], self.master_control, self._optimization_problem)
+        self.__control_field = None
+
+        self.__obj_val = None
+        self.opt_iter = None
+
+    def GetMinimumBufferSize(self) -> int:
+        return 2
+
+    def Check(self):
+        pass
+        
+    def Initialize(self):
+        self.converged = False
+        self.__obj_val = None
+
+        CallOnAll(self.master_control.GetListOfControls(), Control.Initialize)
```

Comment:
```suggestion
        self.master_control.Initialize()
```
Controls are already initialized from the `optimization_analysis`